src/main/java/com/commercetools/sync/services/CustomObjectService.java

```diff
@@ -0,0 +1,90 @@
+package com.commercetools.sync.services;
+
+import com.commercetools.sync.customobjects.helpers.CustomObjectCompositeIdentifier;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+public interface CustomObjectService {
+
+    /**
+     * Given a {@code identifier} and convert the object to string as key, this method first checks if a cached
```

Comment:
The keyToId cached map in `BaseService` class requires `Map<String, String>`. Therefore we convert the identifier object to String by using `.toString()` so that it can act as a key in the cached map.

Reply:
So the method is doing it or is this conversion also given? It's not clear from the comment.

Reply:
The conversion is doing in `CustomObjectServiceImpl`